src/esp/gfx/PTexMeshShader.cpp

```diff
@@ -87,5 +90,36 @@ PTexMeshShader& PTexMeshShader::bindAdjFacesBufferTexture(
   return *this;
 }
 
+PTexMeshShader& PTexMeshShader::setMVPMatrix(const Magnum::Matrix4& matrix) {
+  setUniform(MVPMatrixUniform_, matrix);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setExposure(float exposure) {
+  setUniform(exposureUniform_, exposure);
+  return *this;
+}
+PTexMeshShader& PTexMeshShader::setGamma(float gamma) {
+  setUniform(gammaUniform_, gamma);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setSaturation(float saturation) {
+  setUniform(saturationUniform_, saturation);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setAtlasTextureSize(
+    Magnum::GL::Texture2D& texture,
+    uint32_t tileSize) {
+  setUniform(tileSizeUniform_, (int)tileSize);
+
+  // get image width in given mip level 0
+  int mipLevel = 0;
+  const auto width = texture.imageSize(mipLevel).x();
```

Comment:
Since the rendering uses geometry shaders, I assume it's not planned to be running on WebGL, right? (If it would be, the `imageSize()` would be problematic, but otherwise not at all.)

Reply:
@msbaines: May I ask you what our latest plan is on rendering the ptex mesh on WebGL?

Reply:
@mosra : Yeah, the geometry shader is a problem as it is not available in WebGL.

Reply:
I am currently planning on using the vertex color information from the ply mesh but at some point would like to migrate to using the ptex textures. Not sure when. It will depend on whether the vertex color-only is OK. The other issue is that the textures are 100s of MB each so may not be feasible for WebGL.

Reply:
Texture size could be fixed by using Basis texture compression (see https://github.com/mosra/magnum-plugins/pull/62, I'm in process of integrating that), ideally the compressed textures being upstreamed back to Replica.

For vertex colors, when I checked how a PLY file looks compared to the textured GLB, it was quite underwhelming :) For human eyes and a 4K screen at least, when rendering tiny images the difference would probably be minimal.